sky/server/daemons.py

```diff
@@ -67,6 +69,11 @@ def run_event(self):
                     sky_logging.reload_logger()
                     level = self.refresh_log_level()
                     self.event_fn()
+                # Clear request level cache after each run to avoid
+                # using too much memory.
+                for func in annotations.FUNCTIONS_NEED_RELOAD_CACHE:
+                    func.cache_clear()
```

Comment:
nit: Seems like we should just make a function in annotations module that does this.